src/object_store/src/object/s3.rs

```diff
@@ -207,24 +201,13 @@ impl StreamingUploader for S3StreamingUploader {
             "s3 write bytes error"
         )));
         let data_len = data.len();
-        self.not_uploaded_len += data_len;
+        self.next_part_len += data_len;
         self.buf.push(data);
 
-        if self.not_uploaded_len > self.part_size {
-            if self.part_end == 0 {
-                // Mark current slice of buffer to be the next part to be uploaded.
-                self.part_end = self.buf.len();
-            } else {
-                // `data` should be uploaded in the next part.
-                self.next_part_len += data_len;
-            }
-        }
-        if self.next_part_len >= MIN_PART_SIZE {
+        if self.next_part_len > self.part_size {
```

Comment:
Seems that after this PR, we cannot ensure that the last part can have at least `MIN_PART_SIZE`.

Reply:
I do not understand, if the last part does not meet this condition, we can not upload it?

Reply:
The last part is uploaded in the same API as other part, and in such API we didn't find any way to specify that this is the last part. Therefore, in the perspective of S3, the last part is a normal part with no difference to other parts, and the part size limit should still be guaranteed.

Reply:
I have revert it to main branch